### PR TITLE
IMPLEMENTED DECIMAL TO BINARY NUMBER CONVERSION PROGRAM 

### LINKED ISSUE #4342: DECIMAL TO BINARY CONVERSION USING STACK IN C
DECIMAL TO BINARY CONVERSION USING STACK IN C
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes !

AIM :: **DECIMAL TO BINARY CONVERSION USING STACK IN C**

## Motivation

As i didn't found DECIMAL TO BINARY CONVERSION USING STACK IN C, So i am willing to contribute it...

## Pitch

Hi I am GSSoC'21 participant,
"I am looking to work on the DECIMAL TO BINARY CONVERSION USING STACK IN C"
I will write the clean and optimized code , by keeping best case time and space complexity in mind and 

I will use C language.

### PR DESCRIPTION
**Pull request by ananddasani**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes !!

### Description

I have implemented the Decimal to Binary number conversion problem 

I request @ankitaggarwal23 , @HarshCasper to please review my PR.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

No !!

Fixes #4342

### CODE REVIEW

File: C/README.md [hunk 1]
```diff
@@ -172,6 +172,8 @@ _add list here_
 - [Happy Number](math/happy_number.c)
 - [Automorphic Number](math/automorphic_no.c)
 - [Karpekar Number](math/Kaprekar_no.c)
+- [Neon Number](math/Neon_number.c) 
+- [Decimal To Binary] (math/DecimalToBinary.c)
```
Comment (ankitaggarwal23) [inline C/README.md]: Remove the space between `]  (math`

File: C/README.md [hunk 2]
```diff
@@ -172,6 +172,8 @@ _add list here_
 - [Happy Number](math/happy_number.c)
 - [Automorphic Number](math/automorphic_no.c)
 - [Karpekar Number](math/Kaprekar_no.c)
+- [Neon Number](math/Neon_number.c)
```
Comment (ankitaggarwal23) [inline C/README.md]: Remove this `- [Neon Number](math/Neon_number.c) ` why you are adding this as the file already present Found three similar line kind remove the remaining two.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C/README.md (hunk 1 above)
Comment (ananddasani) [reply C/README.md]: okay

**Comment by ananddasani** [comment]
@ankitaggarwal23 okay i got ,, i have deleted that 2 more neonumber list from README file at the time of solving conflicts ,
now you can check everything is alright i think :)

File: C/math/DecimalToBinary.c [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+//for insertion
+void push(int);
+
+//for deletion
+int pop();
+
+//to check stack is empty or full
+int isEmpty();
+int isFull();
```
Comment (ankitaggarwal23) [inline C/math/DecimalToBinary.c]: Remove the function declaration.

File: C/math/DecimalToBinary.c [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    printf("\nDECIMAL :: %d\n", number);
+    printf("BINARY  :: ");
+
+    while (!isEmpty())
+        printf("%d", pop());
+
+    return 0;
+}
```
Comment (ankitaggarwal23) [inline C/math/DecimalToBinary.c]: Shift the int main() after all the function end.

File: C/math/DecimalToBinary.c [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+int isEmpty()
```
Comment (ankitaggarwal23) [inline C/math/DecimalToBinary.c:16]: Add a one line description above the function.

File: C/math/DecimalToBinary.c [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+{
+    if (top == -1)
+        return 1;
+
+    return 0;
+}
+
+int isFull()
```
Comment (ankitaggarwal23) [inline C/math/DecimalToBinary.c:25]: Add a one line description above the function.

File: C/math/DecimalToBinary.c [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+{
+    if (top == SIZE - 1)
+        return 1;
+
+    return 0;
+}
+
+void push(int r)
```
Comment (ankitaggarwal23) [inline C/math/DecimalToBinary.c:34]: Add a one line description above the function.

File: C/math/DecimalToBinary.c [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+{
+    if (isFull())
+        printf("\nSTACK OVERFLOW\n");
+    else
+        stack[++top] = r;
+}
+
+int pop()
```
Comment (ankitaggarwal23) [inline C/math/DecimalToBinary.c:43]: Add a one line description above the function.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by ananddasani** [comment]
@ankitaggarwal23 done :)

**Review by ankitaggarwal23 — APPROVED** [review]

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
@ananddasani Why you are adding Neon Number file again. Kindly delete it
Capture

**Comment by ananddasani** [comment]
@ankitaggarwal23  now see !! 

don't know how this neon number is being added again again 🤷‍♂️😂

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**